airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/JsonSchemaPrimitive.java

```diff
@@ -8,6 +8,7 @@ public enum JsonSchemaPrimitive {
 
   STRING,
   NUMBER,
+  INTEGER,
```

Comment:
@airbytehq/protocol-reviewers can someone take a look into this change?

Reply:
@marcosmarxm Out of curiosity, why do we need an integer type when it already has `NUMBER` that encompasses integer values?  See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Number.html for more details.

Reply:
Hey @jdpgrailsdev I have added this as a first class JsonSchemaPrimitive for the reason that Number is used for storing both decimal and integeter [numbers](https://json-schema.org/understanding-json-schema/reference/numeric.html#number) and in the IncrementalUtil.compareCursor it is used for comparing doubles instead of integers. Since Dynamodb can support incremental syncs using epoch  timestamps I needed the ability to compare two integers for the state trasmission.

Reply:
@itaseskii You can get an integer value from a `Number`, so I'm not sure why we would need an explicit `INTEGER` value in the protocol.  I'm also not sure why it would matter if the code is comparing two `Number` objects.  Could you point out where in the pull request you are doing this comparison or where in the existing code you think it would be an issue to compare two `Number` objects instead of integers?

Reply:
@jdpgrailsdev The implementation in `IncrementalUtil.compareCursor()` is not comparing Numbers but Doubles when a `JsonSchemaPrimitive.Number` is provided. But even so there is a point that it shouldn't be an issue comparing two integers using `Double.compare(Double.parseDouble(original), Double.parseDouble(candidate))` since they can easily be parsed to doubles if the value is provided as String. I have reverted the addition of the Integer type to the JsonSchemaPrimitive enum and thanks a lot for your input!